parsers/US_SPP.py

```diff
@@ -296,37 +301,32 @@ def fetch_wind_solar_forecasts(
         solar = _NaN_safe_get(forecast, "Solar Forecast MW")
         wind = _NaN_safe_get(forecast, "Wind Forecast MW")
 
-        production = {}
-        if solar is not None:
-            production["solar"] = solar
-        if wind is not None:
-            production["wind"] = wind
-
-        if production == {}:
+        if solar is None and wind is None:
             logger.info(
                 f"fetch_wind_solar_forecasts: {dt} has no solar nor wind forecasted production"
             )
             continue
 
-        datapoint = {
-            "datetime": dt,
-            "production": production,
-            "zoneKey": zone_key,
-            "source": "spp.org",
-        }
+        mix = ProductionMix(solar=solar, wind=wind)
```

Comment:
I noticed the source can return the following:
```python
{'correctedModes': [], 'datetime': datetime.datetime(2024, 3, 19, 7, 0, tzinfo=datetime.timezone.utc), 'production': {'solar': nan, 'wind': nan}, 'source': 'spp.org', 'sourceType': <EventSourceType.forecasted: 'forecasted'>, 'storage': {}, 'zoneKey': 'US-CENT-SWPP'}
```
I'll double check where we want to handle this with the rest of the team and see if we want to handle this in the parser or more likley in the parser classes directly. I'll put this on hold until then but so far it looks good. Should have an answer by tomorrow on how we want to handle it.

Reply:
I have created https://github.com/electricitymaps/electricitymaps-contrib/pull/6558 as a fix, will do a final review of this after that is merged.